AmplifyPlugins/Analytics/AWSPinpointAnalyticsPlugin/Dependency/Pinpoint/Analytics/AnalyticsClient.swift

```diff
@@ -0,0 +1,95 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+import StoreKit
+
+class AnalyticsClient: InternalPinpointClient {
```

Comment:
do we need reference types and inheritance here? can the `InternalPinpointClient` be a protocol with a requirement on the initializer `init(context: PinpointContext)`?

Reply:
I went with the common parent class to enforce having an `unowned` context that is required on init, and to avoid having to implement it on all the clients.

If we go with the protocol we will need to manually define that in all three classes. It's not that much work, but i went for simplicity.